libs/providers/growthbook-client/src/lib/growthbook-client-provider.ts

```diff
@@ -0,0 +1,77 @@
+import { EvaluationContext, Provider, JsonValue, ResolutionDetails, GeneralError } from '@openfeature/web-sdk';
+
+import { Context, GrowthBook } from '@growthbook/growthbook';
+
+export class GrowthbookClientProvider implements Provider {
+  metadata = {
+    name: GrowthbookClientProvider.name,
+  };
+
+  readonly runsOn = 'client';
+  private _client?: GrowthBook;
+
+  hooks = [];
+
+  private get client(): GrowthBook {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
+  }
+
+  // the global context is passed to the initialization function
+  async initialize(context?: EvaluationContext): Promise<void> {
+    // Convert EvaluationContext to growthbook Context
+    this._client = new GrowthBook(context as Context);
+
+    await this._client.loadFeatures();
+  }
+
+  resolveBooleanEvaluation(flagKey: string, defaultValue: boolean): ResolutionDetails<boolean> {
+    const value = this.client.getFeatureValue(flagKey, defaultValue);
+    const featureResult = this.client.evalFeature(flagKey);
+
+    const resolution: ResolutionDetails<boolean> = {
+      value,
+      reason: featureResult.source,
+      variant: featureResult.experimentResult?.variationId?.toString(),
+    };
+    return resolution;
+  }
+
+  resolveStringEvaluation(flagKey: string, defaultValue: string): ResolutionDetails<string> {
+    const value = this.client.getFeatureValue(flagKey, defaultValue);
+    const featureResult = this.client.evalFeature(flagKey);
```

Comment:
Only call `evalFeature` and use logic from within `getFeatureValue ` for `defaultValue` passed in